test-tools/src/main/java/org/terracotta/org/junit/rules/ExternalResource.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Terracotta, Inc., a Software AG company.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.org.junit.rules;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.MultipleFailureException;
+import org.junit.runners.model.Statement;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Equivalent of {@link org.junit.rules.ExternalResource}, but with the following changes:
+ * <ul>
+ *   <li>before and after methods are called with the test {@link Description}</li>
+ *   <li>A failure in the after method won't hide the test failure</li>
+ * </ul>
+ */
+public interface ExternalResource extends TestRule {
```

Comment:
Yes because it does not extend Junit's `ExternalResource`.  We could make it extend Junit `ExternalResource` by rewriting all its implementation, locking (final) the `ExternalResource ` after before and provide our own ones with the test description as parameter.